greenmail-core/src/main/java/com/icegreen/greenmail/imap/commands/SortCommand.java

```diff
@@ -0,0 +1,326 @@
+                            break;
+                        case CC:
+                            compareResult = doCompare(getCc(m1), getCc(m2));
+                            break;
+                        case DATE:
+                            compareResult = doCompare(m1.getMimeMessage().getSentDate(), m2.getMimeMessage().getSentDate());
+                            break;
+                        case FROM:
+                            compareResult = doCompare(getFrom(m1), getFrom(m2));
+                            break;
+                        case SIZE:
+                            compareResult = doCompare(m1.getMimeMessage().getSize(), m2.getMimeMessage().getSize());
+                            break;
+                        case SUBJECT:
+                            compareResult = doCompare(m1.getMimeMessage().getSubject(), m2.getMimeMessage().getSubject());
+                            break;
+                        case TO:
+                            compareResult = doCompare(getTo(m1), getTo(m2));
+                            break;
+                        default:
+                            break;
+                    }
+                    if (compareResult != 0) {
+                        break;
+                    }
+                }
+                return compareResult;
+            } catch (MessagingException e) {
+                throw new RuntimeException(e.getMessage(), e);
+            }
+        }
+
+        private String getFrom(StoredMessage message) throws MessagingException {
+            Address[] addresses = message.getMimeMessage().getFrom();
+            if (addresses == null || addresses.length == 0) {
+                return null;
+            }
+            return String.valueOf(addresses[0]);
+        }
+
+        private String getTo(StoredMessage message) throws MessagingException {
+            Address[] addresses = message.getMimeMessage().getRecipients(Message.RecipientType.TO);
+            if (addresses == null || addresses.length == 0) {
+                return null;
+            }
+            return String.valueOf(addresses[0]);
+        }
+
+        private String getCc(StoredMessage message) throws MessagingException {
+            Address[] addresses = message.getMimeMessage().getRecipients(Message.RecipientType.CC);
+            if (addresses == null || addresses.length == 0) {
+                return null;
+            }
+            return String.valueOf(addresses[0]);
+        }
+
+        private int doCompare(Comparable c1, Comparable c2) {
+            int multiplier = reverse.getAndSet(false) ? -1 : 1;
+            if (c1 == c2) {
+                return 0;
+            } else if (c1 == null) {
+                return multiplier * 1;
+            } else if (c2 == null) {
+                return multiplier * -1;
+            }
+            return multiplier * c1.compareTo(c2);
+        }
+    }
+
+
+    private class SortCommandParser extends CommandParser {
```

Comment:
This should be split out into its own public class. This inner class is difficult to unit test.